src/actions/Activator.ts

```diff
@@ -55,26 +60,32 @@ export class Activator implements IExecutor {
         MqttProvider.publishEvent('fail', ['Activator', 'execute'], 'Missing/invalid WSMan response payload', clientObj.uuid)
         throw new RPSError(`Device ${clientObj.uuid} activation failed. Missing/invalid WSMan response payload.`)
       } else if (clientObj.activationStatus.activated && clientObj.activationStatus.changePassword) {
-        const result = await updateAMTAdminPassword(clientId, wsmanResponse, this.amtwsman, this.clientManager, this.configurator, this.validator)
-        if (result) {
+        const result = await updateAMTAdminPassword(clientId, wsmanResponse, this.responseMsg, this.clientManager, this.configurator, this.validator, httpHandler)
+        if (result.method === 'success') {
```

Comment:
what if result is not defined?

Reply:
It will throw an exception and returns an error message to AMT.